pkg/chunk/cassandra/storage_client.go

```diff
@@ -81,13 +84,15 @@ func (cfg *Config) session() (*gocql.Session, error) {
 			Max:        cfg.MaxBackoff,
 		}
 	}
-	cfg.setClusterConfig(cluster)
+	if err = cfg.setClusterConfig(cluster); err != nil {
+		return nil, errors.WithStack(err)
+	}
 
 	return cluster.CreateSession()
 }
 
 // apply config settings to a cassandra ClusterConfig
-func (cfg *Config) setClusterConfig(cluster *gocql.ClusterConfig) {
+func (cfg *Config) setClusterConfig(cluster *gocql.ClusterConfig) error {
```

Comment:
Please add a unit test on this function to cover:

- No auth
- Auth with password
- Auth with password file contain just the password
- Auth with password file contain the password  and a newline at the end of the file